igor-web/src/main/groovy/com/netflix/spinnaker/igor/build/InfoController.groovy

```diff
@@ -35,67 +35,80 @@ import org.springframework.web.bind.annotation.ResponseStatus
+    BuildMasters buildMasters
+
+    @Autowired
+    TravisProperties travisProperties
 
     @RequestMapping(value = '/masters', method = RequestMethod.GET)
     List<Object> listMasters(@RequestParam(value = "showUrl", defaultValue = "false") String showUrl) {
         log.info('Getting list of masters')
         if (showUrl == 'true') {
-            jenkinsProperties.masters.collect {
+            List<Object> masterList = jenkinsProperties.masters.collect {
                 [
                     "name"   : it.name,
                     "address": it.address
                 ]
             }
+            masterList.addAll(
+                travisProperties.masters.collect {
+                    [
+                        "name": it.name,
+                        "address": it.address
+                    ]
+                }
+            )
+            return masterList
         } else {
-            masters.map.keySet().sort()
+            List<Object> masterList = buildMasters.map.keySet().sort()
+            masterList //TODO: return raw?
         }
     }
 
     @RequestMapping(value = '/jobs/{master}', method = RequestMethod.GET)
     List<String> getJobs(@PathVariable String master) {
         log.info('Getting list of jobs for master: {}', master)
 
-        def jenkinsService = masters.map[master]
-        if (!jenkinsService) {
-            throw new MasterNotFoundException("Master '${master}' does not exist")
-        }
-
-        def jobList = []
-        def recursiveGetJobs
+        def jenkinsService = buildMasters.filteredMap(BuildServiceProvider.JENKINS)[master]
+        def otherService = buildMasters.map[master]
+        if (jenkinsService){
+            def jobList = []
+            def recursiveGetJobs
 
-        recursiveGetJobs = { list, prefix="" ->
-            if (prefix) {
-                prefix = prefix + "/job/"
-            }
-            list.each {
-                if (it.list == null || it.list.empty) {
-                    jobList << prefix + it.name
-                } else {
-                    recursiveGetJobs(it.list, prefix + it.name)
+            recursiveGetJobs = { list, prefix="" ->
+                if (prefix) {
+                    prefix = prefix + "/job/"
+                }
+                list.each {
+                    if (it.list == null || it.list.empty) {
+                        jobList << prefix + it.name
+                    } else {
+                        recursiveGetJobs(it.list, prefix + it.name)
+                    }
                 }
             }
-        }
-        recursiveGetJobs(jenkinsService.jobs.list)
+            recursiveGetJobs(jenkinsService.jobs.list)
 
-        return jobList
+            return jobList
+        } else if (otherService) {
+            return buildCache.getJobNames(master)
+        } else {
+            throw new MasterNotFoundException("Master '${master}' does not exist")
+        }
     }
 
     @RequestMapping(value = '/jobs/{master}/**')
```

Comment:
Added in squashed commit. Thanks for the heads up, I had not spotted this endpoint.